src/panel/PanelManager.jsx

```diff
@@ -212,12 +216,41 @@ export default class PanelManager extends React.Component {
   }
 
   render() {
-    const { ActivePanel, options, panelSize } = this.state;
+    const { ActivePanel, options, panelSize, isPanelVisible } = this.state;
+    const { searchBarInputNode, searchBarOutputNode } = this.props;
 
-    return <PanelContext.Provider value={{ size: panelSize, setSize: this.setPanelSize }} >
-      <div className="panel_container">
-        <ActivePanel {...options} />
-      </div>
-    </PanelContext.Provider>;
+    return <div>
+      <Suggest
+        inputNode={searchBarInputNode}
+        outputNode={searchBarOutputNode}
+        withCategories
+        onChange={query => {
+          if (ActivePanel === ServicePanel) {
+            this.setState({ isPanelVisible: query.length === 0 });
```

Comment:
The race conditions are still present. `<Suggest>` will suffer from unnecessary renders on successive input events while the active panel is ServicePanel.